### DISCUSSION TITLE
SOPS encrypted secret in flux-system namespace

### DISCUSSION THREAD

**Discussion by tingold** [fluxcd/flux2#2256] [Q&A] [answered]
Hey all, 

I'm trying to add a container registry secret to the `flux-system` namespace so that it can be referenced from multiple `ImageRepository` declarations within that namespace.

The issue is that no matter what I do, the secret is never decrypted via SOPS (AWS KMS).  I've successfully configured SOPS on all non `flux-system` namespaces and they work great. 

It appears that decryption is not enabled by default in the `Kustomization` found in `gotk-sync.yaml`. Adding it in a decryption block (I saw the DO NOT EDIT header) seems to get ignored.  

I'm guessing that adding another `source` and `kustomization` targeting the `flux-system` namespace might work...but it feels wrong to have 2 sources contributing config to the same namespace. 

Thoughts?

**Comment by stefanprodan** [comment, marked as answer]
You don't need another source, you can create a Kustomization for your secrets that use the `flux-system` source.  Make sure those secrets are outside of the cluster dir so that the root Kustomization does not apply them.

**Reply by tingold** [reply]
Got it, that makes sense. Thanks!
